### ISSUE TITLE
8.56-based Images fail to build

### ISSUE THREAD

**Issue by iversond** [psadmin-io/ps-vagabond#53]
HR Image 23 (and ELM Image 16) are based on PeopleTools 8.56 and the DPK has changed so the images fail to build (Windows and Linux).

**Comment by iversond** [comment]
Read the `bs-manifest` file under `DPK_INSTALL\setup` to determine which DPK commands to use? Or make it an option in `config.rb`?

**Comment by iversond** [comment]
Here is a 
```powershell 
(((select-string -path .\bs-manifest -Pattern "version=(.*)").matches.groups[1].value).replace('.','')).substring(0,3)
```

**Comment by jrbing** [comment]
OK, I was _finally_ able to get this sorted out for the Linux provisioning script.  Here's a rundown of what's changed.  

- The script now checks the `bs-manifest` file to determine the tools version before proceeding
- For 8.55 based DPK's: 
  - Run the `psft-dpk-setup.sh` script with the `--no_env_setup` option
  - Copy the `psft_customizations.yaml` file into the `$PUPPET_HOME/data` directory
  - Execute `puppet apply $PUPPET_HOME/manifests/site.pp` to apply the DPK
- For 8.56 based DPK's:
  - Generate a response file at `$DPK_INSTALL/response.cfg` containing default credentials (these can still be overridden in `psft_customizations.yaml`
  - Run the `psft-dpk-setup.sh` script with the `--customization_file=$CUSTOMIZATION_FILE` option and the and the `--response_file $DPK_INSTALL/response.cfg` option

I tried using the `--no_puppet_run` option for the 8.56 DPK (which is basically the same as --no_env_setup), but it looks like there's a bug in the `psft_environment.py` script at line 108 which causes it to error out.  It appears to be attempting to generate and execute a DMS script which modifies the `PS_PTIASPDMPARAM` table (multilingual settings).  Since `puppet apply` hasn't been run, the database doesn't yet exist and the script fails.  This wouldn't otherwise be an issue (prior 8.55 PUM images appear to fail on this step as well), except that it prevents the script from deploying Elasticsearch (which is now included in 8.56 DPK releases).

**Comment by iversond** [comment]
This has been resolved and working well.
